flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/codegen/calls/ScalarOperatorGens.scala

```diff
@@ -168,8 +168,15 @@ object ScalarOperatorGens {
         generateBinaryArithmeticOperator(ctx, op, left.resultType, left, right)
 
       case (DATE, INTERVAL_DAY_TIME) =>
-        generateOperatorIfNotNull(ctx, new DateType(), left, right) {
-          (l, r) => s"$l $op ((int) ($r / ${MILLIS_PER_DAY}L))"
+        resultType.getTypeRoot match {
```

Comment:
What is for?
Why in `[FLINK-13561][table-planner-blink] Drop TO_TIMESTAMP(bigint) function`?

Reply:
It is in order to fix `timestampadd(hour, 1, date '2019-08-08')` which will get `2019-08-08` as result.

I'm sorry I messed it in `to_timestamp` commit. I should split it out.